pkg/providerconfig/types.go

```diff
@@ -16,32 +16,37 @@ limitations under the License.
 
 package providerconfig
 
-import "k8s.io/apimachinery/pkg/runtime"
-
-type CloudProvider string
-
-const (
-	CloudProviderAWS          CloudProvider = "aws"
-	CloudProviderAzure        CloudProvider = "azure"
-	CloudProviderDigitalocean CloudProvider = "digitalocean"
-	CloudProviderGoogle       CloudProvider = "gce"
-	CloudProviderHetzner      CloudProvider = "hetzner"
-	CloudProviderKubeVirt     CloudProvider = "kubevirt"
-	CloudProviderLinode       CloudProvider = "linode"
-	CloudProviderOpenstack    CloudProvider = "openstack"
-	CloudProviderPacket       CloudProvider = "packet"
-	CloudProviderVsphere      CloudProvider = "vsphere"
-	CloudProviderFake         CloudProvider = "fake"
-	CloudProviderAlibaba      CloudProvider = "alibaba"
-	CloudProviderAnexia       CloudProvider = "anexia"
-	CloudProviderScaleway     CloudProvider = "scaleway"
-	CloudProviderBaremetal    CloudProvider = "baremetal"
-	CloudProviderExternal     CloudProvider = "external"
+import (
+	osmv1alpha1 "k8c.io/operating-system-manager/pkg/crd/osm/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// DNSConfig contains a machine's DNS configuration
+type DNSConfig struct {
+	Servers []string `json:"servers"`
+}
+
+// NetworkConfig contains a machine's static network configuration
+type NetworkConfig struct {
+	CIDR    string    `json:"cidr"`
+	Gateway string    `json:"gateway"`
+	DNS     DNSConfig `json:"dns"`
+}
+
 type Config struct {
 	SSHPublicKeys []string `json:"sshPublicKeys"`
+	CAPublicKey   string   `json:"caPublicKey"`
```

Comment:
Why this is added here? Actually the whole Config struct can removed and use the Config type from machine controller type directly:
https://github.com/kubermatic/machine-controller/blob/0bb1509e8e1097430c41cdbd73ad310456706907/pkg/providerconfig/types/types.go#L110-L125